protocol/tx_test.go

```diff
@@ -1,4 +1,4 @@
-package protocol
```

Comment:
Is this move to `protocol_test` necessary? If we really need to move it to a test package, then we should stop using `newTestChain` and use `prottest.NewChain`. `newTestChain` exists only to circumvent with the dependency cycle.